src/App.js

```diff
@@ -1,24 +1,36 @@
-import logo from './logo.svg';
 import './App.css';
+import { useEffect, useState } from 'react';
+import fetchData from './services/blog';
+import PostList from './components/PostList';
 
 function App() {
+  const [posts, setPosts] = useState([])
+  const [hasError, setHasError] = useState(false)
+  const [isLoading, setIsLoading] = useState()
```

Comment:
I believe `false` would be a good initial state

```suggestion
  const [isLoading, setIsLoading] = useState(false)
```